shared/elements/gaia_sim_picker/script.js

```diff
@@ -51,6 +53,21 @@ window.GaiaSimPicker = (function(win) {
     }
   };
 
+  /**
+   * Localize the component manually as l10n attributes are not supported
+   * within the shadow dom. See also: bug 1026236.
+   */
+  proto._localizeShadowDom = function() {
+    var elements = this.shadowRoot.querySelectorAll('[data-l10n-id]');
+    [].forEach.call(elements, el => {
+      var attributes = navigator.mozL10n.getAttributes(el);
+      navigator.mozL10n.formatValue(attributes.id, attributes.args)
+      .then(value => {
+        el.textContent = value;
+      });
+    });
+  };
+
```

Comment:
Oh yeah, definitely, `translateFragment` will sanitize HTML in translations and also properly set localizable attributes like `placeholder` and `aria-*`.

There's also `mozL10n.formatEntity` which returns both the value and the attributes, but it doesn't sanitize HTML and requires some renaming (`ariaLabel` needs to become `aria-label` for instance).  Again, `translateFragment` takes care of all that.